cmake/OpenCVDetectInferenceEngine.cmake

```diff
@@ -94,3 +94,15 @@ if(INF_ENGINE_TARGET)
     INTERFACE_COMPILE_DEFINITIONS "HAVE_INF_ENGINE=1;INF_ENGINE_RELEASE=${INF_ENGINE_RELEASE}"
   )
 endif()
+
+if(WITH_NGRAPH)
+  if(INF_ENGINE_RELEASE VERSION_LESS "2019040000")
+    message(WARNING "nGraph is not tested with InferenceEngine versions prior 2019R4")
+  endif()
+  find_package(ngraph QUIET)
+  if(ngraph_FOUND)
+    ocv_assert(TARGET ngraph::ngraph)
+    message(STATUS "Detected ngraph: cmake package (${ngraph_VERSION})")
+    set(HAVE_NGRAPH ON)
+  endif()
+endif()
```

Comment:
Please also add the following changes to this file: https://github.com/opencv/opencv/commit/087824e57e11fd29ef1caa30334a387348403024#diff-da800785713bed07d12ee19660fa794a (please ignore the changes at modules/dnn/CMakeLists.txt).

It's necessary to use custom location of Inference Engine.